File: Python/Math-Array-String/44.py
# 1773. Count Items Matching a Rule


from typing import List

class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        # Map ruleKey to its respective index in the items list
        key_index_map = {"type": 0, "color": 1, "name": 2}
        rule_index = key_index_map[ruleKey]
        
        # Count the number of matches
        match_count = sum(1 for item in items if item[rule_index] == ruleValue)
        return match_count

sol = Solution()
output = sol.countMatches([["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]], ruleKey = "type", ruleValue = "phone")
print(output)

File: JavaScript/Math-Array-String/44.js
// 1773. Count Items Matching a Rule


/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
    // Map ruleKey to its respective index in the items array
    const keyIndexMap = { "type": 0, "color": 1, "name": 2 };
    const ruleIndex = keyIndexMap[ruleKey];
    
    // Count the number of matches
    let matchCount = 0;
    for (const item of items) {
        if (item[ruleIndex] === ruleValue) {
            matchCount++;
        }
    }
    
    return matchCount;
};

// Example usage:
const items1 = [["phone", "blue", "pixel"], ["computer", "silver", "lenovo"], ["phone", "gold", "iphone"]];
const ruleKey1 = "color";
const ruleValue1 = "silver";
console.log(countMatches(items1, ruleKey1, ruleValue1)); // Output: 1

const items2 = [["phone", "blue", "pixel"], ["computer", "silver", "phone"], ["phone", "gold", "iphone"]];
const ruleKey2 = "type";
const ruleValue2 = "phone";
console.log(countMatches(items2, ruleKey2, ruleValue2)); // Output: 2
